Initialize climber code + make existing more efficient and readable

MathNerd28 opened the pull request:
Make existing code more efficient and readable, add a Climber subsystem, and add a method to square while keeping the sign

MathNerd28 committed:
Initial Climber commit

Make existing code more efficient and readable, add a Climber subsystem, and add a method to square while keeping the sign

mmarchetti commented on .vscode/settings.json:
```diff
@@ -25,5 +25,6 @@
   ],
-  "java.test.defaultConfig": "WPIlibUnitTests"
+  "java.test.defaultConfig": "WPIlibUnitTests",
+  "java.jdt.ls.vmargs": "-XX:+UnlockExperimentalVMOptions"
```
If this flag was added automatically by the upgrade to 2022.2.1, then it's all good. If we added this then we should probably think twice.

MathNerd28 replied:
I had to add this myself to get the language server to start, otherwise I had no language support whatsoever. The new update added a VM argument, and I suspect WPILIB forgot to install the additional argument when upgrading the project.

mmarchetti commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -1,23 +1,19 @@
+    public static final int PILOT_JOYSTICK   = 0;
+    public static final int COPILOT_JOYSTICK = 1;
+
+    public static final double FULL_PRESS_MARGIN = 0.99D;
```
Seems like these 3 constants should live in OperatorInterface.java, since that is where they are used.

MathNerd28 replied:
I don’t know who put them there, I will move them to the OI

mmarchetti commented on src/main/java/frc/robot/DTButton.java:
```diff
@@ -1,39 +1,40 @@
 
 import edu.wpi.first.wpilibj.Joystick;
-//Mike, this one's for you to handle
+// Mike, this one's for you to handle
```
remove?

MathNerd28 replied:
Looks like someone just copied last year’s file and forgot to remove that line... (gone now)

mmarchetti commented on src/main/java/frc/robot/DTButton.java:
```diff
@@ -1,39 +1,40 @@
-        this.button = button;
+    public DTButton(Joystick joystick, int buttonID) {
+        this.joystick = joystick;
+        this.buttonID = buttonID;
```
We haven't standardized on the use of `this` for local fields. Most of our code doesn't use it, so I would suggest going the other way (remove all uses of `this` except where required, e.g. to pass a reference to this into a method of another class).

MathNerd28 replied:
All instances removed

MathNerd28 replied:
(Except the two you highlighted there because changing them would require variable name changes that reduce clarity)

mmarchetti commented on src/main/java/frc/robot/OperatorInterface.java:
```diff
@@ -1,115 +1,172 @@
 /*----------------------------------------------------------------------------*/
-/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
-/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* Copyright (c) 2021-2022 FIRST. All Rights Reserved. */
```
In general, we probably shouldn't change copyright notices unless they are ours.

MathNerd28 replied:
Fixed it, but changed the line endings slightly (so the formatter doesn’t mess with them)

mmarchetti commented on src/main/java/frc/robot/OperatorInterface.java:
```diff
@@ -1,115 +1,172 @@
+
+    // optional
+    private DTButton[]    cocopilotButtons;
+    private final boolean cocopilotButtonsActive;
```
We don't have a cocopilot so probably can drop these and the methods that use them.

mmarchetti commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -8,99 +8,115 @@
+     * You can add additional auto modes by adding additional comparisons to the
+     * switch structure below with additional strings. If using the
+     * SendableChooser make sure to add them to the chooser code above as well.
+     */
```
this comment seems to be obsolete.

MathNerd28 replied:
Removed

mmarchetti reviewed (changes requested):
As a general note, we should avoid systemic formatting changes because they tend to cause merge conflicts with other team members. But since it is early in the season and others mostly aren't modifying this code right now, it's probably ok. 

I'm not sure if the formatting changes here were done manually or via the code formatter in VSCode. I would recommend that everyone run the code formatter on their code before committing - it establishes a readable standard for indentation and other formatting and eliminates the need for a lot of manual formatting.

MathNerd28 committed:
Applied a consistent formatting scheme

Created a formatter profile for our repo (formatter-settings.xml) and included it in the repo. Now, anyone who uses the format command from the command palette will automatically apply a consistent profile across all java files.

MathNerd28 committed:
Fixed problems and stuff

mmarchetti commented:
If anyone has feedback on a PR, please leave feedback (with gracious professionalism) in the comments.

mmarchetti commented:
Or, if these changes were incorporated in the other PR, leaving a comment when closing would be helpful for communicating what's happening.

Rylan-Meilutis commented:
This was incorporated into the other pull request.
